network/src/node.rs

```diff
@@ -294,15 +297,14 @@ impl<S: Storage + Send + core::marker::Sync + 'static> Node<S> {
             .expect("local address was set more than once!");
     }
 
-    pub fn initialize_metrics(&self) {
+    pub async fn initialize_metrics(&self) -> Result<()> {
         debug!("Initializing metrics");
         let metrics_task = snarkos_metrics::initialize();
         self.register_task(metrics_task);
 
         // The node can already be at some non-zero height.
-        if let Some(sync) = self.sync() {
-            metrics::counter!(misc::BLOCK_HEIGHT, sync.current_block_height() as u64);
-        }
+        metrics::counter!(misc::BLOCK_HEIGHT, self.storage.canon().await?.block_height as u64);
```

Comment:
The node might not have a sync layer, meaning it might not have meaningful values related to blocks; this also means it should still be bound to `Sync`, as opposed to always be available in the `Node`.

Reply:
`Sync` doesn't define our block height, storage does. `Sync` provides mutation helpers.